targets/TARGET_Freescale/TARGET_MCUXpresso_MCUS/TARGET_MCU_K64F/mbed_crc_api.c

```diff
@@ -0,0 +1,125 @@
+#include "crc_api.h"
+
+#include "drivers/fsl_crc.h"
+#include "platform/mbed_assert.h"
+
+#ifdef DEVICE_CRC
+
+static crc_bits_t width;
+
+bool hal_crc_is_supported(const uint32_t polynomial)
+{
+  switch (polynomial)
+  {
+    case POLY_8BIT_CCITT:
+    case POLY_7BIT_SD:
+    case POLY_16BIT_CCITT:
+    case POLY_16BIT_IBM:
+    case POLY_32BIT_ANSI:
+      return true;
+    default:
+      return false;
+  }
+}
+
+void hal_crc_compute_partial_start(const uint32_t polynomial)
+{
+  crc_config_t config;
+
+  switch (polynomial)
+  {
+    case POLY_32BIT_ANSI:
+    {
+      width = kCrcBits32;
+
+      config.polynomial         = polynomial;
+      config.seed               = 0xFFFFFFFFU;
+      config.reflectIn          = true;
+      config.reflectOut         = true;
+      config.complementChecksum = true;
+      config.crcBits            = width;
+      config.crcResult          = kCrcFinalChecksum;
+
+      break;
+    }
+    case POLY_16BIT_IBM:
+    {
+      width = kCrcBits16;
+
+      config.polynomial         = polynomial;
+      config.seed               = 0;
+      config.reflectIn          = true;
+      config.reflectOut         = true;
+      config.complementChecksum = false;
+      config.crcBits            = width;
+      config.crcResult          = kCrcFinalChecksum;
+
+      break;
+    }
+    case POLY_16BIT_CCITT:
+    {
+      width = kCrcBits16;
+
+      config.polynomial         = polynomial;
+      config.seed               = 0xFFFFFFFFU;
+      config.reflectIn          = false;
+      config.reflectOut         = false;
+      config.complementChecksum = false;
+      config.crcBits            = width;
+      config.crcResult          = kCrcFinalChecksum;
+
+      break;
+    }
+    case POLY_8BIT_CCITT:
+    {
+      width = kCrcBits16;
+
+      config.polynomial         = polynomial;
+      config.seed               = 0U;
+      config.reflectIn          = false;
+      config.reflectOut         = false;
+      config.complementChecksum = false;
+      config.crcBits            = width;
+      config.crcResult          = kCrcFinalChecksum;
+
+      break;
+    }
+    case POLY_7BIT_SD:
+    {
+      width = kCrcBits16;
+
+      config.polynomial         = polynomial;
+      config.seed               = 0U;
+      config.reflectIn          = false;
+      config.reflectOut         = false;
+      config.complementChecksum = false;
+      config.crcBits            = width;
+      config.crcResult          = kCrcFinalChecksum;
+
+      break;
+    }
+  }
+
+  CRC_Init(CRC0, &config);
+}
+
+void hal_crc_compute_partial(const uint8_t *data, const size_t size)
+{
+  CRC_WriteData(CRC0, data, size);
```

Comment:
I suggest to add robust checks against null pointer and 0 size and add the following information in the requirements section:

`Function hal_crc_compute_partial() does nothing if pointer to buffer is undefined (NULL) or data length is equal to 0`.

On different platforms data size equal to 0 might have different consequences, so we can handle such case and simply return without any action.